src/builder/create_attachment.rs

```diff
@@ -107,15 +137,10 @@ impl<'a> CreateAttachment<'a> {
     }
 }
 
-#[derive(Clone, Debug, Serialize)]
-struct ExistingAttachment {
-    id: AttachmentId,
-}
-
 #[derive(Clone, Debug)]
-enum NewOrExisting<'a> {
+enum EditAttachmentsInner<'a> {
     New(CreateAttachment<'a>),
-    Existing(ExistingAttachment),
+    Existing(AttachmentId),
```

Comment:
Do these changes need to be done as part of this PR? If not, can this be split out to make this PR easier to review and test.

Reply:
Sure, I can separate it out into a separate PR.